File: greedy.c
#include <math.h>
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    float dollars_change;
    
    int change, coins, quarters = 25, dimes = 10, nickels = 5, pennies = 1;
    coins = 0;
    
    printf("O hai! ");
    do
    {
        printf("How much change is owed? ");
        dollars_change = get_float();
    }
    while (dollars_change<=0);
    
    dollars_change = dollars_change*100;
    dollars_change = round(dollars_change);
    
    change = (int)dollars_change;
    
    {
        
    while (change>=quarters)
    {
        change = change - quarters;
        coins++;
    }
    
    while (change>=dimes)
    {
        change = change - dimes;
        coins++;
    }
    
    while (change>=nickels)
    {
        change = change - nickels;
        coins++;
    }
    
    while (change>=pennies)
    {
        change = change - pennies;
        coins++;
    }
    
     printf("%d\n", coins);
    
    }
    
    return 0;
}

File: greedy.py
import cs50 as cs

def main():
    print("O hai! ", end="")
    while True:
        print("How much change is owed?")
        money = cs.get_float()
        if money >= 0:
            break
    money = round(money, 2)*100
    coins = money//25          #for quarters
    remainder = money % 25
    coins += remainder//10     #for dimes
    remainder = money % 10
    coins += remainder//5      #nickels
    remainder = money % 5
    coins += remainder//1       #pennies
    remainder = money % 1
    print(coins)

if __name__=="__main__":
    main()